js/core/utils.js

```diff
@@ -14,6 +14,61 @@ define(
                 msg.pub("end", "core/utils");
                 cb();
             }
+        ,   normalizePadding: function (text) {
+                function testRegex(regex) {
+                    return function(text) {
+                        return regex.test(text);
+                    }
+                }
+                if (!text) {
+                    return "";
+                }
+                if (typeof text !== "string") {
+                    throw TypeError("Invalid input");
+                }
+                var hasOpeningPre = testRegex(/<pre\b[^>]*>/i);
```

Comment:
Conceptually hostile to the idea of recreating your own mini HTML parser using regexes here given you're in a browser and can leverage the native one. Maybe that's the onlu sane solution, though.